db/python/tables/sample.py

```diff
@@ -97,6 +97,78 @@ async def update_sample(
         await self.connection.execute(_query, {**values, 'id': id_})
         return id_
 
+    async def merge_samples(
+        self,
+        id_keep: int = None,
+        id_merge: int = None,
+        author: str = None,
+    ):
+        """Merge two samples together"""
+        sid_merge = sample_id_format(id_merge)
+        _, sample_keep = await self.get_single_by_id(id_keep)
+        _, sample_merge = await self.get_single_by_id(id_merge)
+        merged_from = to_db_json({'merged_from': sid_merge})
+
+        def list_merge(l1: Any, l2: Any) -> List:
+            lst: List = [l1, l2]
+            if isinstance(l1, list) and isinstance(l2, list):
+                lst = l1 + l2
+            elif isinstance(l1, list):
+                lst = l1 + [l2]
+            elif isinstance(l2, list):
+                lst = [l1] + l2
+            return lst
+
+        def dict_merge(meta1, meta2):
+            d = dict(meta1)
+            d.update(meta2)
+            for key, value in meta2.items():
+                if key not in meta1.keys() or meta1[key] is None or value is None:
+                    continue
+
+                d[key] = list_merge(meta1[key], value)
+
+            return d
+
+        values: Dict[str, Any] = {
+            'id': id_keep,
+            'author': author or self.author,
+            'meta': to_db_json(dict_merge(sample_keep.meta, sample_merge.meta)),
+            'merged_from': merged_from,
+        }
+
+        _query = """
+            UPDATE sample
+            SET author = :author,
+                meta = JSON_MERGE_PATCH(:meta, :merged_from)
+            WHERE id = :id
+        """
+        _query_seqs = f"""
+            UPDATE sample_sequencing
+            SET sample_id = {id_keep}
```

Comment:
Have to use a placeholder for these queries and substitute using the second param of `connection.execute`.